ccp/src/cu/proving_thread/sync/channels_facade.rs

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2024 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use ccp_randomx::cache::Cache;
+use ccp_randomx::Dataset;
+use ccp_shared::types::LogicalCoreId;
+
+use super::to_utility_message::ToUtilityInlet;
+use super::to_utility_message::ToUtilityMessage;
+use super::STResult;
+use crate::cu::proving_thread::messages::*;
+use crate::hashrate::ThreadHashrateRecord;
+use crate::utility_thread::message::ProvingThreadSyncError;
+use crate::utility_thread::message::RawProof;
+
+#[derive(Clone, Debug)]
+pub(crate) struct ToAsync {
+    to_async: SyncToAsyncInlet,
+}
+
+impl ToAsync {
+    pub(crate) fn from_channel(to_async: SyncToAsyncInlet) -> Self {
```

Comment:
This is the only constructor, and typically such trivial constructors should be called `new`.